db/snapshot/backup_task.cc

```diff
@@ -97,7 +98,12 @@ future<> backup_task_impl::do_backup() {
         // Parallelism is implicitly controlled in two ways:
         //  - s3::client::claim_memory semaphore
         //  - http::client::max_connections limitation
-        std::ignore = _client->upload_file(component_name, destination, _progress, &_as).handle_exception([comp = component_name, &ex] (std::exception_ptr e) {
+        std::ignore = _client->upload_file(component_name, destination, _progress, &_as).then([component_name] {
+            // Delete the uploaded component to:
+            // 1. Free up disk space immediately
+            // 2. Avoid costly S3 existence checks on future backup attempts
+            return remove_file(component_name.native());
+        }).handle_exception([comp = component_name, &ex] (std::exception_ptr e) {
             snap_log.error("Error uploading {}: {}", comp.native(), e);
```

Comment:
This .handle_exception will collect exceptions from both -- upload_file() and remove_file() -- and it might be hard to tell which part failed exactly